### PR TITLE
feat(generate): add guard generation

### PR DESCRIPTION
**Pull request by delasteve**

### CODE REVIEW

File: packages/angular-cli/lib/config/schema.json [hunk 1]
```diff
@@ -234,6 +234,10 @@
               "type": "boolean",
               "default": true
             },
+            "guard": {
+              "type": "boolean",
```
Comment (hansl) [inline packages/angular-cli/lib/config/schema.json]: Description field please.

**Comment by delasteve** [comment]
@hansl, could I get a re-review? Thanks!

**Comment by delasteve** [comment]
@hansl could I get a rebuild on `travis` and `appveyor`? Please and thank you!

**Comment by filipesilva** [comment]
@delasteve there seems to be some broken tests with your latest changes.

**Pull request merged**